File: 232-implement-queue-using-stacks/implement-queue-using-stacks.java
class MyQueue {
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public MyQueue() {
        stack1 = new Stack<>();
        stack2 = new Stack<>();
    }

    public void push(int x) {
        stack1.push(x);
    }

    public int pop() {
        int x = this.peek();
        return stack2.pop();
    }

    public int peek() {
        if (stack2.isEmpty()) {
            // copy all
            while(stack1.isEmpty()==false) stack2.add(stack1.pop());
        }
        return stack2.peek();
    }

    public boolean empty() {
        return stack1.isEmpty() && stack2.isEmpty();
    }
}

File: 232-implement-queue-using-stacks/implement-queue-using-stacks.py
class MyQueue:
    def __init__(self):
        self.stack1=[]
        self.stack2=[]
        
    def push(self, val:int):
        self.stack1.append(val)
        
    def peek(self) -> int:
        if not self.stack2:
            while self.stack1: self.stack2.append(self.stack1.pop())
        return -1 if not self.stack2 else self.stack2[-1]
        
    def pop(self) -> int:
        if not self.stack2:
            while self.stack1:
                self.stack2.append(self.stack1.pop())
        return self.stack2.pop() if self.stack2 else -1

    def empty(self):
        return not self.stack1 and not self.stack2
         


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()